src/node/mod.rs

```diff
@@ -13,35 +13,69 @@ use channel::network_channel;
 pub use controller::{Controller, Packet};
 pub use header::Header;
 
+use crate::node::state::State;
 pub use state::{MemoryState, SledDiskState};
+use std::net::SocketAddr;
+use std::str::FromStr;
 use std::sync::Arc;
 use tokio::net::TcpStream;
 use tokio::sync::Mutex;
 use tracing::info;
 pub use wire::Wire;
 
-pub async fn node_with_single_peer(address: &str) -> anyhow::Result<()> {
+pub async fn node_with_autodiscovery(addresses_override: Option<String>) -> anyhow::Result<()> {
     let network = Network::Live;
     // let state = SledDiskState::new(Network::Live);
     let state = MemoryState::new(network);
 
     let state = Arc::new(Mutex::new(state));
-    let address = address.to_owned();
-
-    // TODO: peering.nano.org
-
-    let _state_clone = Arc::clone(&state);
-    info!("Spawning a channel to {}", &address);
-    let handle = tokio::spawn(async move {
-        let stream = TcpStream::connect(&address).await.unwrap();
-        // let mut channel = Channel::new(network, state_clone, stream).await;
-        // channel.run().await.unwrap();
-        network_channel(network, state, stream)
+    let configured_peers = if addresses_override.is_some() {
+        parse_socket_list(addresses_override.unwrap())
```

Comment:
Let's remove all these `unwrap()`s. I know the previous code here had them too, but the single node connection was just a quick hacky implementation to connect to my own node.

I'd suggest using `.context()?` or `.with_context()?` depending if you need to dynamically `format!()` a string.

Reply:
This particular unwrap is the only one I would not remove since it's the only one that should never fail, as in: if it fails there was a programming error, since the `if` checks exactly for that.
All the others I'll try to make them manage the error better.

Reply:
Ah yes sorry, I didn't notice the `is_some()`.

A safer way of doing this to prevent the chance of a programming error in the future is: `if let Some(addrs) = addresses_override {}` which will do the `is_some()` check as well as an `unwrap()` in one go.

Reply:
Ah yes that's a good suggestion. I've done a mix of the two because otherwise it would become tricky to pass the result to `add_peers`